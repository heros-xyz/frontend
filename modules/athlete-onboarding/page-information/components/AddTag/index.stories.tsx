import type { Meta, StoryObj } from "@storybook/react";
import AddTag from ".";

const meta: Meta<typeof AddTag> = {
  title: "Athlete/AddTag",
  component: AddTag,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof AddTag>;

export const AddTagStory: Story = {};
