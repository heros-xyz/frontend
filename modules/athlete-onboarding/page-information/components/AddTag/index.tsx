import { Box, Flex, Text, Input } from "@chakra-ui/react";
import { ChangeEvent, FC, KeyboardEvent, useEffect, useState } from "react";
import { Close } from "@/components/svg/Close";
import HerosOnboardingWrapper from "@/components/ui/HerosOnboardingWrapper";
import ErrorMessage from "@/components/common/ErrorMessage";

interface AddTagProps {
  onSubmit: (value: string[]) => void;
}

const AddTag: FC<AddTagProps> = ({ onSubmit }) => {
  const [tags, setTags] = useState<string[]>([]);
  const [inputValue, setInputValue] = useState("");

  const [isInvalid, setIsInvalid] = useState(false);

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    if (e.target.value === ",") return;
    setInputValue(e.target.value);
  };

  const handleDeleteTag = (index: number) => {
    setTags((prev) => [...prev].filter((_, idx) => idx !== index));
  };

  const handleKeyDown = (e: KeyboardEvent) => {
    if ((e.key === "Enter" || e.key === ",") && Boolean(inputValue)) {
      if (inputValue.length > 25) {
        setIsInvalid(true);
        return;
      }
      if (inputValue.length < 25 && inputValue[0] === "#") {
        setTags((prev) => [
          ...prev,
          inputValue.trim().replace(/[^a-zA-Z ]/g, ""),
        ]);
        setInputValue("");
        setIsInvalid(false);
        return;
      }
    }
  };

  return (
    <HerosOnboardingWrapper
      onSubmit={() => onSubmit(tags)}
      textButton="SUBMIT"
      title="PAGE INFORMATION"
    >
      <Box>
        <Text as="p" fontWeight={500} fontSize="base" mb={2.5}>
          Add Tag(s)
        </Text>
        <Text as="p" fontSize="xs" fontWeight={400} mb={2.5}>
          Help people find you easier by adding keywords related to you and your
          content. Tags will not be visible to fans on your profile page.
          Separate tags by comma.
        </Text>
        <Text fontSize="12px" as="p" mb="20px" fontWeight={600}>
          Ideas for you: #sport, #athlete,...
        </Text>
        <Input
          variant="flushed"
          mb={2.5}
          placeholder="Add Tags"
          borderColor="primary"
          fontSize={{ base: "sm" }}
          onKeyDown={handleKeyDown}
          value={inputValue}
          onChange={handleChange}
        />
        <ErrorMessage
          condition={isInvalid}
          errorMessage="Tagline cannot exceed 25 characters."
        />
        <Flex flexWrap="wrap">
          {tags.map((it, index) => (
            <Text
              key={it}
              as="span"
              bg="accent.2"
              textColor="white"
              py="3px"
              px="12px"
              rounded="full"
              fontSize="14px"
              fontWeight={500}
              mr="12px"
              mb="6px"
            >
              #{it}{" "}
              <Close
                width="9px"
                height="9px"
                ml="10px"
                cursor="pointer"
                onClick={() => handleDeleteTag(index)}
              />
            </Text>
          ))}
        </Flex>
      </Box>
    </HerosOnboardingWrapper>
  );
};

export default AddTag;
