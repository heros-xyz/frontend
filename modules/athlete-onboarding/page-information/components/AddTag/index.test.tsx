import { render } from "@testing-library/react";
import AddTag from ".";

it("renders add tag unchanged", () => {
  const { container } = render(<AddTag onSubmit={console.log} />);
  expect(container).toMatchSnapshot();
});
