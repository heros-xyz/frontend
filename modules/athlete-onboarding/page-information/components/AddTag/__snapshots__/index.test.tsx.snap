// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders add tag unchanged 1`] = `
<div>
  <div
    class="css-1jvt6v8"
  >
    <div
      class="css-1kmmdcq"
    >
      Page information
    </div>
    <div
      class="chakra-container css-17f3xev"
    >
      <div
        class="css-1iltqwc"
      >
        <div
          class="css-14s5ckc"
        >
          <div
            class="css-1s2o82u"
          >
            <div
              class="css-5k9psj"
            >
              <div
                class="css-11pajvj"
              >
                <p
                  class="chakra-text css-12j3vcb"
                >
                  Add Tag(s)
                </p>
                <p
                  class="chakra-text css-1suvgez"
                >
                  Help people find you easier by adding keywords related to you and your content. Tags will not be visible to fans on your profile page. Separate tags by comma.
                </p>
                <p
                  class="chakra-text css-sphssm"
                >
                  Ideas for you: #sport, #athlete,...
                </p>
                <input
                  class="chakra-input css-1dhk8tj"
                  placeholder="Add Tags"
                  value=""
                />
                <div
                  class="css-2imjyh"
                />
              </div>
              <div
                class="css-lrby83"
              >
                <button
                  class="chakra-button css-k6pd2s"
                  type="button"
                >
                  SUBMIT
                  <span
                    class="css-9n3bbc"
                  />
                </button>
              </div>
            </div>
          </div>
          <button
            class="chakra-button css-1mv50kv"
            type="button"
          >
            SUBMIT
            <span
              class="css-9n3bbc"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
